### ISSUE TITLE
FutureRestore error 44 on a Valid APTicket

### ISSUE THREAD

**Issue by dallasgroot** [encounter/futurerestore#38]
Hey there, first off, here is my debug [log from future restore](https://pastebin.com/ridLtRNA)

I have an iPhone 8 10,4 D201AP 
I've verified my APTicket (shsh2) file using 1conans online checker and comes out [valid](https://pastebin.com/UiTJACzw) 

when I run the following command 
`./futurerestore -d -t blobs.shsh2 --latest-baseband -p BuildManifest.plist -s sep-firmware.d201.RELEASE.im4p -m BuildManifest.plist 11.3.1.ipsw`Futurerestore will start doing its thing past the nonce and fails when it checks the apticket telling me it isn't valid, when indeed it is...

I searched through the BuildManifest plist file and moved my device to to top and still no luck.... anyone able to help me or know of a fix for this issue?

iPhone 8 11.3b6 - 11.3.1
iOS 11.3b6 Sep
iOS 11.3.1 ipsw

**Comment by xybp888** [comment]
Me Too!
iPhone 8 Plus !
Tell me how to solve this problem,Please!

**Comment by PassiveModding** [comment]
Same here. Debug info:
```
checking APTicket to be valid for this restore...
Verified ECID in APTicket matches device ECID
checking APTicket to be valid for this restore...
Verified ECID in APTicket matches device ECID
[Error] findAnyBuildidentityForFilehash: can't find digest for key=SE,Bootloader. i=1
[Error] im4m_buildidentity_check_cb: can't find any identity which matches all hashes inside IM4M
[Error] getBuildIdentityForIM4M: found buildidentiy, but can't read information
[Error] BuildIdentity selected for restore does not match APTicket

BuildIdentity selected for restore:
BuildNumber : 15E302
BuildTrain : Emet
DeviceClass : d20ap
FDRSupport : YES
RestoreBehavior : Erase
Variant : Customer Erase Install (IPSW)

BuildIdentiy valid for the APTicket:
IM4M is not valid for any restore within the Buildmanifest
This APTicket can't be used for restoring this firmware
[Error] APTicket can't be used for this restore
Done: restoring failed.
Failed with errorcode=-44
```

Using ios 11.2.1 => ios 11.3.1 
Blobs are verified

**Comment by dallasgroot** [comment]
Edit the plist found on the 11.3.1 to just display your devices arrays. Also, what command are you using? 

There are tools for both Mac and windows where you can edit a zip file without decompressing and compressing it again. The problem is that futurerestore checks the BuildManifest.Plist and the first device it sees is iPhone 7. So removing all traces of the iPhone 7 and just focusing on your own device will make futurerestore match your device boardid to your APTicket and come out valid. IPhone 7 didn’t merge with iPhone ipsw intil 11.3 where this all came out.  

Here is a sample BuildManifest.plist that I used for my successful restore. 
https://www.dropbox.com/s/t7gy1qdwp7bexqm/11.3b6.zip?dl=0&m=

Edit both plist files from 11.3b6 and 11.3.1 inside the ipsw. This should work!

**Comment by PassiveModding** [comment]
I'm using `futurerestore -t blob.shsh2 -d --latest-baseband -p 114BuildManifest-Modified.plist -s 114sep-firmware.d20.RELEASE.im4p -m 114sepManifest-Unmodified.plist 1131.ipsw`

Still failing on the same place. Should I be recompiling 1131.ipsw with a modified BuildManifest?

**Comment by dallasgroot** [comment]
Yes you should, and don’t use that command. Use the 11.3b6 files not latest anything :).

**Comment by dallasgroot** [comment]
And then use

./futurerestore -d -t blobs.shsh2 -b baseband_file -p 11.3b6BuildManifest.plist -s 11.3b6sepfile -m BuildManifest.plist 11.3.1.ipsw

**Comment by PassiveModding** [comment]
Ah right, Should I be using the same buildmanifest for both or different ones?
ie.
`futurerestore -d -t blobs.shsh2 -b baseband_file -p 11.3b6BuildManifest.plist -s 11.3b6sepfile -m 11.3b6BuildManifest.plist 11.3.1.ipsw`

Also where should I download the beta ipsw from?

**Comment by dallasgroot** [comment]
No don’t use the same BuildManifest.plist. They should be separate as the last BuildManifest points to 11.3.1ipsw. I’m only putting names for references. Still call your 11.36b BuildManifest.plist it’s original name

**Comment by xybp888** [comment]
please use my Build for windows ver v178
download URL : 
https://bbs-att-qcloud.weiphone.net/2018/06/21/14541769_futurerestore_win64_v178.zip
this Bug fix for @xybp888

**Comment by PassiveModding** [comment]
@xybp888 Was able to get it to actually make it past the error 44, but it ended up failing to install and I've had to update to 11.4 to fix rip

**Comment by PassiveModding** [comment]
Any chance I can go from 11.4 backwards?

**Comment by dallasgroot** [comment]
11.3b6

**Comment by xybp888** [comment]
@PassiveModding  tell Me you log 
If there is an -4 error, please wait patiently!
It will not affect the recovery process

this fix -4 error 
https://bbs-att-qcloud.weiphone.net/2018/06/22/14543572_futurerestore_win64_v178-1.zip
You can now restore 11.3beta6 again
And then try again

this my log.txt 
https://bbs-att-qcloud.weiphone.net/2018/06/21/14541770_log2.txt

**Comment by HaaxXxeD** [comment]
@dallasgroot first of tell me how to recompile ipsw with modified build manifest i change in manifest and compress file and change extension in to ipsw and it said the ipsw is courpted, didn’t find the buildManifest.plist from ipsw. How to solve this error. Thanks.

**Comment by PassiveModding** [comment]
I'm not going to bother trying for now, No reason risking now that 11.3b6 is unsigned
